cmd/root.go

```diff
@@ -17,175 +17,69 @@ package cmd
 import (
 	"errors"
 	"fmt"
+	"io"
 	"os"
-	"path/filepath"
 
-	"github.com/pulumi/crd2pulumi/gen"
+	"github.com/pulumi/crd2pulumi/pkg/codegen"
 	"github.com/spf13/cobra"
-	"github.com/spf13/pflag"
 )
 
-const (
-	DotNet string = "dotnet"
-	Go     string = "go"
-	NodeJS string = "nodejs"
-	Python string = "python"
-)
-
-const (
-	DotNetPath string = "dotnetPath"
-	GoPath     string = "goPath"
-	NodeJSPath string = "nodejsPath"
-	PythonPath string = "pythonPath"
-)
-
-const (
-	DotNetName string = "dotnetName"
-	GoName     string = "goName"
-	NodeJSName string = "nodejsName"
-	PythonName string = "pythonName"
-)
-
-const defaultOutputPath = "crds/"
+// Version specifies the crd2pulumi version. It should be set by the linker via LDFLAGS. This defaults to dev
+var Version = "dev"
 
 const long = `crd2pulumi is a CLI tool that generates typed Kubernetes 
 CustomResources to use in Pulumi programs, based on a
 CustomResourceDefinition YAML schema.`
 
-const example = `crd2pulumi --nodejs crontabs.yaml
-crd2pulumi -dgnp crd-certificates.yaml crd-issuers.yaml crd-challenges.yaml
-crd2pulumi --pythonPath=crds/python/istio --nodejsPath=crds/nodejs/istio crd-all.gen.yaml crd-mixer.yaml crd-operator.yaml
-crd2pulumi --pythonPath=crds/python/gke https://raw.githubusercontent.com/GoogleCloudPlatform/gke-managed-certs/master/deploy/managedcertificates-crd.yaml
-
-Notice that by just setting a language-specific output path (--pythonPath, --nodejsPath, etc) the code will
-still get generated, so setting -p, -n, etc becomes unnecessary.
+const example = `crd2pulumi --lang typescript crontabs.yaml
+crd2pulumi --lang go crd-certificates.yaml crd-issuers.yaml crd-challenges.yaml
+crd2pulumi --lang python --packageName myProject --outputDir=crds/python/istio crd-all.gen.yaml crd-mixer.yaml crd-operator.yaml
+crd2pulumi --lang csharp --outputDir=crds/dotnet/gke https://raw.githubusercontent.com/GoogleCloudPlatform/gke-managed-certs/master/deploy/managedcertificates-crd.yaml
```

Comment:
While I do like the new names better, this is a breaking change. I think we'll need to either release this change as a 2.0, or continue to support the old options as deprecated fallback options.